salt/template.py

```diff
@@ -88,8 +88,9 @@ def compile_template(template,
         if not isinstance(input_data, (dict, OrderedDict)):
             try:
                 input_data.seek(0)
-            except Exception as exp:
-                log.error('error while compiling template \'{0}\': {1}'.format(template, exp))
+            except AttributeError:
```

Comment:
I can see the case for catching and ignoring an `AttributeError` but I don't follow why we would also remove the generalized `Exception` handling. Could you explain why that's necessary here?